ruddock/modules/rotation/routes.py

```diff
@@ -44,3 +50,34 @@ def update_comment(prefrosh_id):
   helpers.update_comments(prefrosh_id, flask.request.form.get("comments", ""))
   helpers.update_votes(prefrosh_id, flask.request.form)
   return flask.redirect(flask.url_for("rotation.show_prefrosh", prefrosh_id=prefrosh_id))
+
+@blueprint.route('/move/', methods=['POST'])
+@login_required(Permissions.ROTATION)
+def move():
+  old_bucket_name = flask.request.form.get("oldBucket")
+  new_bucket_name = flask.request.form.get("newBucket")
+  if old_bucket_name not in helpers.BUCKETS or new_bucket_name not in helpers.BUCKETS:
+    flask.flash("Bad value for one of the buckets.")
+    return flask.redirect(flask.url_for('rotation.show_portal'))
+  elif old_bucket_name == new_bucket_name:
+    flask.flash("Buckets must be distinct")
+    return flask.redirect(flask.url_for('rotation.show_portal'))
+  else:
+    old_bucket_prefrosh = helpers.get_prefrosh_by_bucket(old_bucket_name)
+    new_bucket_prefrosh = helpers.get_prefrosh_by_bucket(new_bucket_name)
+    return flask.render_template('move.html', old_bucket=old_bucket_prefrosh,
+      new_bucket=new_bucket_prefrosh, old_bucket_name=old_bucket_name, new_bucket_name=new_bucket_name,
+      vote_tuples=helpers.VOTE_TUPLES)
+
+@blueprint.route('/change_bucket/<int:prefrosh_id>', methods=['POST'])
+@login_required(Permissions.ROTATION)
+def change_bucket(prefrosh_id):
+  new_bucket_name = flask.request.form.get("newBucket")
+  old_bucket_name = flask.request.form.get("oldBucket")
+  bucket_prefrosh = helpers.get_prefrosh_by_bucket(old_bucket_name)
+  prefrosh, prev_id, next_id = helpers.get_prefrosh_and_adjacent(prefrosh_id, bucket_prefrosh)
+  helpers.change_bucket(prefrosh_id, new_bucket_name)
+  if prev_id:
+    return flask.redirect(flask.url_for("rotation.move", _anchor=prev_id), code=307)
```

Comment:
the default redirect status code (I think it's 302) will convert the request to a get request, but in this case we want the request to stay a post request since `rotation.move` is a post endpoint. point is moot though since we're changing `rotation.move` to be a get endpoint.